test/monkey_patches/route_params.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+# TODO: remove when you see similar error but actual format is: "format"=>:json
+# PARAMS_TO_SYMBOLIZE causes errors in the tests regarding the format.
+# It always converts format to symbol which causes a failure in the tests:
+# Stats::Api::ApplicationsController should route GET /stats/api/applications/42/usage.json
+# --- expected
+# +++ actual
+# @@ -1 +1 @@
+# -{"application_id"=>"42", "action"=>"usage", "format"=>:json, "controller"=>"stats/api/applications"}
+# +{"controller"=>"stats/api/applications", "action"=>"usage", "application_id"=>"42", "format"=>"json"}
+module Shoulda
+  module Matchers
+    module ActionController
+      class RouteParams
+        PARAMS_TO_SYMBOLIZE = [].freeze
+      end
+    end
+  end
+end
```

Comment:
We can't do this in our case. We are using one route for multiple formats and default won't work.
One of the tests that are failing because of this: https://github.com/3scale/porta/blob/master/test/functional/stats/api/services_controller_test.rb#L7


On the other hand, when the tests are checking the controller they are parsing the path and the path contains the format which is parsed as string, not a symbol.
https://github.com/rails/rails/blob/v5.1.7/actionpack/lib/action_dispatch/routing/route_set.rb#L864